### DISCUSSION TITLE
How to send post request with csrf token in a custom action?

### DISCUSSION THREAD

**Discussion by dhr0118** [filamentphp/filament#10733] [Help]
```php   
Action::make('refund'))    
->action(function (Orders $orders) {                          
  $jsonData = [
      '_token'=>csrf_token(),
      'out_trade_no' => $trans_id,
   ];

  $url = route('refund');
  $response = Http::post($url, $jsonData);

  ...})
```

```php
//in routes\web.php

Route::post('/orders/refund',[OrdersController::class,'refundOrder'])->name('refund');
```

response always return error code 419, how to fix it?

**Comment by leandrocfe** [comment]
You could add this route in the `routes/api.php` instead of `web.php`

**Reply by dhr0118** [reply]
Yes, I know it. I just want to know why it doesn't work in the `web.php`. I mean, I've written the csrf token in the request body, right?

**Reply by leandrocfe** [reply]
It makes sense because the token is sent, but I don't know why it returns 419 in the next request.
Another way to avoid this would be add the route in the VerifyCsrfToken file

```php
//app/Http/Middleware/VerifyCsrfToken.php
protected $except = [
    '/orders/refund'
];
```
